supervisor/supervisor.go

```diff
@@ -64,14 +64,46 @@ func setupEventLog(s *Supervisor) error {
 	}
 	logrus.WithField("count", len(s.eventLog)).Debug("containerd: read past events")
 	events := s.Events(time.Time{})
-	f, err := os.OpenFile(filepath.Join(s.stateDir, "events.log"), os.O_WRONLY|os.O_CREATE|os.O_APPEND|os.O_TRUNC, 0755)
+	return eventLogger(s, filepath.Join(s.stateDir, "events.log"), events)
+}
+
+func eventLogger(s *Supervisor, path string, events chan Event) error {
+	f, err := os.OpenFile(path, os.O_WRONLY|os.O_CREATE|os.O_APPEND|os.O_TRUNC, 0755)
 	if err != nil {
 		return err
 	}
-	enc := json.NewEncoder(f)
 	go func() {
+		var (
+			count int
+			enc   = json.NewEncoder(f)
+		)
 		for e := range events {
+			if count > 500 {
+				logrus.Debug("truncating event log file")
+				// close the log file
+				if f != nil {
+					f.Close()
+				}
+				slice := 499
+				l := len(s.eventLog)
+				if slice >= l {
+					slice = l
+				}
+				s.eventLog = s.eventLog[len(s.eventLog)-slice:]
```

Comment:
If I'm not mistaken the first time, assuming `readEventLog()` didn't read a thing, this is going to do `s.eventLog[2:]` (i.e. 501 - 499) where we want `s.eventLog[500:]`.

Also, don't this need to be sync'ed with `s.Events(from time.Time)` now ?